modules/aruco/samples/tutorial_charuco_create_detect.cpp

```diff
@@ -21,7 +21,7 @@ static bool readCameraParameters(std::string filename, cv::Mat& camMatrix, cv::M
         return false;
     fs["camera_matrix"] >> camMatrix;
     fs["distortion_coefficients"] >> distCoeffs;
-    return true;
+    return (camMatrix.size() == cv::Size(3,3) && distCoeffs.size() == cv::Size(5,1));
```

Comment:
`distCoeffs` can be different (contains different number of elements).

I believe we should check this instead: `!distCoeffs.empty()` or even remove that check and keep `camMatrix` validation only.

Reply:
`distCoeffs.empty()` compiles and functions.   I don't know anything about well formed distCoeffs.

If one edits the calib.txt file to an ill-formed 1x3 matrix, you get an informative error like: 

libc++abi.dylib: terminating with uncaught exception of type cv::Exception: OpenCV(3.4.2) /opt/concourse/worker/volumes/live/9523d527-1b9e-48e0-7ed0-a36adde286f0/volume/opencv-suite_1535558719691/work/modules/imgproc/src/undistort.cpp:400: error: (-215:Assertion failed) (((_distCoeffs) != __null && (((const CvMat*)(_distCoeffs))->type & 0xFFFF0000) == 0x42420000 && ((const CvMat*)(_distCoeffs))->cols > 0 && ((const CvMat*)(_distCoeffs))->rows > 0) && ((const CvMat*)(_distCoeffs))->data.ptr != __null) && (_distCoeffs->rows == 1 || _distCoeffs->cols == 1) && (_distCoeffs->rows*_distCoeffs->cols == 4 || _distCoeffs->rows*_distCoeffs->cols == 5 || _distCoeffs->rows*_distCoeffs->cols == 8 || _distCoeffs->rows*_distCoeffs->cols == 12 || _distCoeffs->rows*_distCoeffs->cols == 14) in function 'cvUndistortPointsInternal'

If you edit the calib file to eliminate the distortion_coefficients field, then distCoeffs.empty() is test true, but the program still is able to function and do the detection and pose estimation. 

`camMatrix` validation only is good enough.